src/sync/sync_user.hpp

```diff
@@ -31,6 +31,16 @@ namespace realm {
 
 class SyncSession;
 
+enum class SyncUserAdminMode {
+    // User is not an admin.
+    None,
+    // User wraps an admin token. Such users use the tokens they were configured
+    // with to directly open sessions, and do not make network requests.
+    WrapsAdminToken,
+    // User was marked by the authentication server as a user with admin privileges.
+    MarkedAsAdmin,
+};
```

Comment:
I decided to go with something similar to your second approach.

The sync user has APIs to check whether it's an admin, and to update its admin status (as a result of the auth server sending us a response).

We also make a distinction now between users that are or aren't saved in the metadata database. Right now, the only non-persisted users are admin-token-wrapping users, so the code treats non-persisted users as admin users. Once we move the responsibility of token-wrapping users into the binding (as we discussed earlier), we can make persistence/non-persistence completely orthogonal to admin status, or we can remove support for non-persisted users altogether.

I'm happy to make further changes if this approach isn't satisfactory.